llvm_util/utils.cpp

```diff
@@ -165,6 +170,34 @@ Value* get_operand(llvm::Value *v) {
     return ret;
   }
 
+  if (auto gv = dyn_cast<llvm::GlobalVariable>(v)) {
+    auto itr = globalvars.find(gv);
+    if (itr != globalvars.end())
+      return itr->second;
+
+    if (gv->getValueType()->isArrayTy()) {
+      // TODO: Array type of a global variable is not supported.
```

Comment:
I don't see any reason for limiting the type. In the worst case, get_operand returns a null expr.

Reply:
A global array's type is a pointer to array, so llvm_type2alive(v->getType()) at the beginning of get_operand successfully returns IR type, that's why isArrayTy() here is needed.